test/CMakeLists.txt

```diff
@@ -54,6 +54,44 @@ add_executable(${test_target} ${SPARROW_TESTS_SOURCES})
 target_link_libraries(${test_target} PRIVATE sparrow doctest::doctest)
 add_test(NAME ${test_target} COMMAND ${test_target})
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
```

Comment:
I would like that we try all the flags recommended by Jason Turner: https://github.com/cpp-best-practices/cppbestpractices/blob/master/02-Use_the_Tools_Available.md#compilers
and then progressively remove what we dont want.

For MSVC we also need to add `/Zc:__cplusplus` to make sure the `__cplusplus` define has the right value, see: https://learn.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=msvc-170

Finally, shouldnt we have these flags in cmake toolchains instead of here? Doesnt this will affect users? (not sure what's the best practice regarding this)

Reply:
I activated all warnings, just to test, I finaly deactivated a dozen of them. It's maybe easier to maintain than activating some of them and forgot some.
I let you check the code (I commented the flags to explicitly activate the warnings)

Reply:
About where to put these flags, I think we should keep them in our test as the user may not have the same requirements in terms of warnings as errors